src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,169 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final Object[] EMPTY_DATAARRAY = {};
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] dataArray;
+    private int size;
+
+    public ArrayList() {
+        this.dataArray = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity == 0) {
+            this.dataArray = EMPTY_DATAARRAY;
+        } else if (capacity < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Capacity can't be negative");
+        } else {
+            this.dataArray = new Object[capacity];
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == dataArray.length) {
+            grow(dataArray);
+            addOnNextFreeSell(value);
+        } else {
+            addOnNextFreeSell(value);
+        }
```

Comment:
A small suggestion. We will call `addOnNextFreeSell` in any case, so we can eliminate duplication
```suggestion
        if (size == dataArray.length) {
            grow(dataArray);
        }
        addOnNextFreeSell(value);
```